heon/generate_flow_seq.py

```diff
@@ -0,0 +1,111 @@
+import argparse
+import random
+
+
+def flush_file(file):
+    f = open(file, "w")
+    f.write("")
+    f.close()
+
+
+def write_flow_num_to_file(file, value):
+    f = open(file, "a")
+    f.write(f"{value}\n")
+    f.close()
+
+
+def run_single_iteration(flow_num_list, file_to_write):
+    list_to_choose_from = flow_num_list.copy()
+    for i in range(len(list_to_choose_from)):
+        chosen_num = random.choice(list_to_choose_from)
+        write_flow_num_to_file(file_to_write, chosen_num)
+        list_to_choose_from.remove(chosen_num)
+
+
+def generate_random_flow_sequence_grouped(flow_num_list, num_iter, file_to_write):  
+    for i in range(0, num_iter):
+        run_single_iteration(flow_num_list, file_to_write)
+        
+
+def generate_random_flow_sequence_independent(num_conn, flow_num_list, num_iter, file_to_write):
+    flow_num_remaining_count_dict = {}
+    for i in range(1, num_conn+1):
+        flow_num_remaining_count_dict[i] = num_iter
+        
+    list_to_choose_from = flow_num_list.copy()
+    for i in range(0, num_conn * num_iter):
+        chosen_num = random.choice(list_to_choose_from)
+        write_flow_num_to_file(file_to_write, chosen_num)
+        remaining_count = flow_num_remaining_count_dict[chosen_num]
+        remaining_count = remaining_count - 1
+        flow_num_remaining_count_dict[chosen_num] = remaining_count
+        if remaining_count <= 0:
+            list_to_choose_from.remove(chosen_num)
+
+
+def generate_desired_flow_sequence(flow_num_list, num_iter, file_to_write):
```

Comment:
Right now, this is just emitting the input flow/packet sequence. Is this correct?
It may help to rename the function `fifo_flow_sequence` to make that clear, and then call that function from within `generate_desired_flow_sequence` based on an argument that (by default) is set to fifo. It is OK for now to just support this one choice for the desired order argument; we will add synchronized, etc. later on.

Reply:
@ngsrinivas 
I am not sure I follow. I intended this function to just iterate the flow number list a given number of times, which will write flows in the desired order. The `flow_num_list` will always be in ascending order, so if it is [1, 2, 3] for example, this function will iterate through that list, then repeat that iteration by `num_iter` times, which will look something like [1 ,2, 3, 1, 2, 3, 1, 2, 3, ...]. 

Is this what you meant by fifo?

Reply:
(wrong notion of "desired flow sequence")
(1 1 2 2 3 3 -> 1 3 2 / 1 3 2 OR 2 3 1 / 2 3 1 OR ...) -> (within each batch, order is preserved)
(instead of keeping a desired flow sequence file, what about having a script that goes through the output of the server and validates the order?)